### ISSUE TITLE
Comment API should support GET 

### ISSUE THREAD

**Issue by Sumukh** [Cal-CS-61A-Staff/ok#984]
And maybe PUT 

Useful for overcode

**Comment by Sumukh** [comment]
@MrVantastic would like to work on this. Let's just scope this to getting the comments for a single backup. 

Here's what to look at: 
- server/models.py lists the fields for Comment 
- server/controllers/api.py: 
    - CommentSchema -> you'll want to add a get_fields. Look at some of the ones in other schemas to get an idea of how those are structured 
    - CommentAPI -> add a get method that takes in a backup id and outputs the comments that are available for that backup.
         - Look at some of the other get methods to gete an idea of how these work
- documentation/api.md 
   - Add a description of the new methods (See the changes in here for an example: https://github.com/Cal-CS-61A-Staff/ok/commit/20dc87929420fd58919b544e1d2b1ed07443cd66#diff-053f23b623cca0fc70a15951bdcb8eef) 
- tests/test_api.py 
  - Add a test to make sure that the GET request works. 
  - The tests should also verify that unauthorized users cannot access the comments via the API 


Looking for in the PR: 
 - Permissions are correct and there are tests to verify it 
 - The output data format for the API response makes sense 
 - Documentation added to api.md

**Comment by MrVantastic** [comment]
I've written the code and a function to test it, but I'm having trouble actually testing it. When I try testing using the instructions in SETUP.md, I keep getting the same error for all of the tests that run:
 
```
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/_pytest/python.py:418: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/py/_path/local.py:662: in pyimport
    __import__(modname)
tests/__init__.py:7: in <module>
    from server import create_app
server/__init__.py:8: in <module>
    from server import assets, converters, logging, utils
server/converters.py:3: in <module>
    from server import constants, utils
server/utils.py:19: in <module>
    from pynliner import fromString as emailFormat
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pynliner/__init__.py:35: in <module>
    import cssutils
E     File "/Library/Python/2.7/site-packages/cssutils/__init__.py", line 331
E       except xml.dom.HierarchyRequestErr, e:
E                                         ^
E   SyntaxError: invalid syntax
```

And I'm not sure what's wrong. After googling the error, I suspect that it might be a pathing issue between Python 2.7 and Python 3.6 installed on my computer, but I'm not sure. I can't set up a local server anymore either (it gives me the same error). Any help would be much appreciated! I am running Mac OS Sierra 10.12.3. 

**Context:**
I had a server up and running earlier today and I initially tried testing my code with the command:
`python tests/test_api.py`
but that was before I found SETUP.md file. When I tried that command, it said I was missing packages (Flask, Flask RQ, webassets, etc. A list of these packages can be found in /server/__init__.py, lines 1-6 and /server/utils.py lines 9-23.), so I began installing the missing packages one by one in an effort to dispel the error. However, after a certain extent, I ran into another error:
```
  File "/Users/Jeff/Desktop/dev/okpy/ok/server/utils.py", line 27, in <module>
    logger = logging.getLogger(__name__)
AttributeError: 'module' object has no attribute 'getLogger'
```
After which, I figured I was way too deep into this rabbit hole, and I uninstalled all the packages I had just installed. However, after doing this, I am still running into the "Syntax error" shown above.

**Comment by Sumukh** [comment]
Here's a hack: just file a pull request - your tests will run on our continuous integration server. 
Not the best idea but it's better than nothing 






On Sun, Mar 5, 2017 at 4:26 PM -0800, "Jeffrey Van" <notifications@github.com> wrote:












I've written the code and a function to test it, but I'm having trouble actually testing it. When I try testing using the instructions in SETUP.md, I keep getting the same error for all of the tests that run:
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/_pytest/python.py:418: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/py/_path/local.py:662: in pyimport
    __import__(modname)
tests/__init__.py:7: in <module>
    from server import create_app
server/__init__.py:8: in <module>
    from server import assets, converters, logging, utils
server/converters.py:3: in <module>
    from server import constants, utils
server/utils.py:19: in <module>
    from pynliner import fromString as emailFormat
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pynliner/__init__.py:35: in <module>
    import cssutils
E     File "/Library/Python/2.7/site-packages/cssutils/__init__.py", line 331
E       except xml.dom.HierarchyRequestErr, e:
E                                         ^
E   SyntaxError: invalid syntax



And I'm not sure what's wrong. After googling the error, I suspect that it might be a pathing issue between Python 2.7 and Python 3.6 installed on my computer, but I'm not sure. I can't set up a local server anymore either (it gives me the same error). Any help would be much appreciated! I am running Mac OS Sierra 10.12.3.


Context:

I had a server up and running earlier today and I initially tried testing my code with the command:

python tests/test_api.py

but that was before I found SETUP.md file. When I tried that command, it said I was missing packages (Flask, Flask RQ, webassets, etc. A list of these packages can be found in /server/init.py, lines 1-6 and /server/utils.py lines 9-23.), so I began installing the missing packages one by one in an effort to dispel the error. However, after a certain extent, I ran into another error:
  File "/Users/Jeff/Desktop/dev/okpy/ok/server/utils.py", line 27, in <module>
    logger = logging.getLogger(__name__)
AttributeError: 'module' object has no attribute 'getLogger'



After which, I figured I was way too deep into this rabbit hole, and I uninstalled all the packages I had just installed. However, after doing this, I am still running into the "Syntax error" shown above.
